src/components/ColorPickerContainer.ts

```diff
@@ -0,0 +1,269 @@
+            displayColorPicker: this.state.displayColorPicker,
+            style: !hasLabel ? ColorPickerContainer.parseStyle(this.props.style) : undefined,
+            onChange: this.updateColorValue,
+            onChangeComplete: this.handleOnChange
+        }, this.props.mode === "input"
+                ? createElement(Input, {
+                    disabled: this.disabled,
+                    color: this.state.color,
+                    onChange: this.handleInputChange
+                }, this.renderColorPickerButton())
+                : this.renderColorPickerButton()
+        );
+    }
+
+    private renderColorPickerButton() {
+        return createElement(Button, {
+            className: this.props.mode === "input" ? "widget-color-picker-input-inner" : "widget-color-picker-inner",
+            disabled: this.disabled,
+            mode: this.props.mode,
+            color: this.state.color,
+            ...(!this.disabled && this.props.mode !== "inline")
+                ? { onClick: this.handleClick }
+                : {}
+        });
+    }
+
+    private handleClick = () => {
+        this.setState({ displayColorPicker: !this.state.displayColorPicker });
+    }
+
+    private handleClose = () => {
+        this.setState({ displayColorPicker: false });
+    }
+
+    private updateColorValue = (color: ColorResult) => {
+        const { format, mxObject, colorAttribute } = this.props;
+        if (color && mxObject && !this.disabled) {
+            if (format === "hex") {
+                mxObject.set(colorAttribute, color.hex);
+            } else if (format === "rgb") {
+                mxObject.set(colorAttribute, `rgb(${color.rgb.r},${color.rgb.g},${color.rgb.b})`);
+            } else {
+                mxObject.set(colorAttribute, `rgba(${color.rgb.r},${color.rgb.g},${color.rgb.b},${color.rgb.a})`);
+            }
+
+            this.setState({ alertMessage: "" });
+        }
+    }
+
+    private getValue = (mxObject?: mendix.lib.MxObject): string => {
+        if (mxObject) {
+            return mxObject.get(this.props.colorAttribute) as string;
+        }
+
+        return "";
+    }
+
+    private resetSubscriptions(mxObject?: mendix.lib.MxObject) {
+        this.subscriptionHandles.forEach(window.mx.data.unsubscribe);
+        this.subscriptionHandles = [];
+
+        if (mxObject) {
+            this.subscriptionHandles.push(window.mx.data.subscribe({
+                attr: this.props.colorAttribute,
+                callback: this.handleSubscriptions,
+                guid: mxObject.getGuid()
+            }));
+            this.subscriptionHandles.push(window.mx.data.subscribe({
+                callback: this.handleSubscriptions,
+                guid: mxObject.getGuid()
+            }));
+        }
+    }
+
+    private handleSubscriptions = () => {
+        this.setState(this.validateColor(this.props.mxObject));
+    }
+
+    private validateColor = (mxObject?: mendix.lib.MxObject): ColorPickerContainerState => {
```

Comment:
Don't think there's much value here unless u find a way to make it perfect... if intent on keeping it, at least improve it to check for the required characters as the start of the colour string.

Reply:
will look into the reg